sale_import_base/models/sale_channel_importer.py

```diff
@@ -0,0 +1,296 @@
+            result["active"] = False
+        if data.get("country_code"):
+            country = self.env["res.country"].search(
+                [("code", "=", data["country_code"])]
+            )
+            if not country:
+                raise ValidationError(
+                    _("Missing country {}").format(data["country_code"])
+                )
+            result["country_id"] = country.id
+            if data.get("state_code"):
+                state = self.env["res.country.state"].search(
+                    [("code", "=", data["state_code"]), ("country_id", "=", country.id)]
+                )
+                if not state:
+                    raise ValidationError(
+                        _("Missing State %(state_code)s for country %(country_name)s")
+                        % {
+                            "state_code": data["state_code"],
+                            "country_name": country.name,
+                        }
+                    )
+                result["state_id"] = state.id
+        return result
+
+    def _should_merge_addresses(self, vals_addr_invoice, vals_addr_shipping):
+        return vals_addr_invoice == vals_addr_shipping
+
+    def _process_addresses(
+        self, parent, address_invoice, address_shipping, archive_addresses=True
+    ):
+        vals_addr_invoice = self._prepare_partner(
+            address_invoice, parent.id, archive_addresses
+        )
+        vals_addr_shipping = self._prepare_partner(
+            address_shipping, parent.id, archive_addresses
+        )
+        if self._should_merge_addresses(vals_addr_invoice, vals_addr_shipping):
+            # not technically correct for the shipping addr, but this shouldn't matter
+            vals_addr_invoice["type"] = "invoice"
+            result = self.env["res.partner"].create(vals_addr_invoice)
+            return (result, result)
+        else:
+            vals_addr_invoice["type"] = "invoice"
+            vals_addr_shipping["type"] = "delivery"
+            return (
+                self.env["res.partner"].create(vals_addr_invoice),
+                self.env["res.partner"].create(vals_addr_shipping),
+            )
+
+    def _prepare_sale_line_vals(self, data, sale_order):
+        return [self._prepare_sale_line(line, sale_order) for line in data["lines"]]
+
+    def _prepare_sale_line(self, line_data, sale_order):
+        channel = self.payload_id.sale_channel_id
+        company_id = channel.company_id
+
+        product = self.env["product.product"].search(
+            [
+                ("default_code", "=", line_data["product_code"]),
+                ("product_tmpl_id.company_id", "=", company_id.id),
+            ]
+        )
+        if not product:
+            product = self.env["product.product"].search(
+                [
+                    ("default_code", "=", line_data["product_code"]),
+                    ("product_tmpl_id.company_id", "=", False),
+                ]
+            )
```

Comment:
One more thing @florian-dacosta in a recent use case I needed to override the way the product is searched (searching on the barcode instead of searching on default_code), so I created this dedicated method on the v14 sale_import_base : https://github.com/akretion/sale-import/commit/7422ad73fbd4788ca54672cfcfa2d38ba3f66c33#diff-27e4483cdc08344f76a62e3859deaafd9e5ac59ddedb1a44d839c17b3b4c0b77R211

Do you think you can create this method here too?
Thanks!

Reply:
Make sense to search on barcode indeed. I wonder if this should not be configurable on channel or something.
Anyway, wouldn't it be enough to make a hook to get the domain [("default_code", "=", line_data["product_code"])] ?

It seems enough and easier to override.
Something like 
```
base_product_domain = self._get_product_domain(line_data)
product = self.env["product.product"].search(
          base_product_domain + 
              [("product_tmpl_id.company_id", "=", company_id.id)]
      )

def _get_product_domain(self, line_data):
    return [("default_code", "=", line_data["product_code"])] 

```

Reply:
Yes it might be better that way! I'll change my commit on the v14

Reply:
Hum right, I did not think about this.
Well maybe it is best to do as you did then. (We do not want multiple override for such a simple change...)
I'd only rename the `get_product_id` by `get_product` and same for `product_id` variable by `product`
if you don't mind, because I think it is more consistent to the way variable are defined globally in Odoo. (usually with `get_product_id` you would expect an id, and not a recordset) Well, not mandatory, I guess it is also a question of habit/preference, but it seems weird to me !